jobs.groovy.j2

```diff
@@ -644,6 +672,25 @@ branches.each {
             failUnstable(true)
             failNoReports(false)
           }
+{# Update the commit status on GitHub with the build result                  #}
+          flexiblePublish {
+            condition {
```

Comment:
Can you had a link to the plugin page for future reference?: https://wiki.jenkins-ci.org/display/JENKINS/Flexible+Publish+Plugin 

and another for the conditions block?
https://wiki.jenkins-ci.org/display/JENKINS/Run+Condition+Plugin

the flexiblePublish is not very well documented upstream (well most jenkins plugins have little documention, but this one is a bit worse than most). 
The job DSL api:
https://jenkinsci.github.io/job-dsl-plugin/#method/javaposse.jobdsl.dsl.helpers.publisher.PublisherContext.flexiblePublish
states that the build step capability used below is part of the 'any build steps' plugin.
https://wiki.jenkins-ci.org/display/JENKINS/Any+Build+Step+Plugin